flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/sink/filesystem/Buckets.java

```diff
@@ -304,6 +304,10 @@ void close() {
 	}
 
 	private Path assembleBucketPath(BucketID bucketId) {
+		String child = bucketId.toString();
+		if ("".equals(child)) {
+			return basePath;
+		}
 		return new Path(basePath, bucketId.toString());
```

Comment:
Now that you have given `bucketId.toString()` to the variable `child`, you can change it to: `return new Path(basePath, child);`